File: backend/src/main/java/com/urmyfood/backend/domain/model/Post.java
package com.urmyfood.backend.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Post {
    private UUID postId;
    private String dishName;
    private BigDecimal price;
    private BigDecimal originalPrice;
    private int maxQuantity;
    private int remainingQuantity;
    private OffsetDateTime endTime;
    private boolean isFlashSale;
    private PostStatus status;
    private String content;
    private String imageUrl;
    private String category;
    private Account author;
    private OffsetDateTime createdAt;
}

File: frontend/app-shop/src/main/java/com/urmyfood/shop/domain/model/Post.kt
package com.urmyfood.shop.domain.model

data class Post(
    val postId: String,
    val dishName: String,
    val price: Long,
    val originalPrice: Long?,
    val maxQuantity: Int,
    val remainingQuantity: Int,
    val endTime: String?,
    val isFlashSale: Boolean,
    val status: String,
    val content: String?,
    val imageUrl: String?,
    val shopName: String,
    val shopAvatarUrl: String?,
    val createdAt: String,
    val likeCount: Long,
    val isLiked: Boolean,
    val commentCount: Long,
    val category: String?
) {
    val isActive: Boolean get() = status == "ACTIVE"
    val stock: Int get() = remainingQuantity
    val maxStock: Int get() = maxQuantity
}
